openlr_dereferencer/decoding/scoring.py

```diff
@@ -27,16 +27,65 @@ def score_geolocation(wanted: LocationReferencePoint, actual: PointOnLine, radiu
         return 1.0 - dist / radius
     return 0.0
 
+
+def angle_sector(angle: float) -> int:
+    """The bearing angles are mapped to one of  32 sectors, each of size 11.5°
+    returns the sector the angle belongs to.
+
+    Args:
+        angle:
+            the value is expected in degrees
+    Returns:
+        the sector to which the angle belongs to. Value in range [0,31]
+    """
+
+    if angle < 0:
+        angle = angle + 360
+    return int(angle / (360 / 32)) % 32
+
+
+def angle_sector_difference(angle1: float, angle2: float) -> int:
+    """"The distance of the two sectors containing the angles values, respectively
+    Args:
+        angle1, angle2:
+            the values are expected in degrees
+    Returns:
+        Value in the range [0,16]
+        """
+
+    sector_diff = abs(angle_sector(angle1) - angle_sector(angle2))
+    "Differences should be between 0 and 16. Direction (clockwise or counter clockwise) between the two sectors should "
+    "not matter. Thus map a difference (sector_diff) larger than 16 needs to be mapped to 32-sector_diff"
+    if sector_diff > 16:
+        sector_diff = 32 - sector_diff
+    return sector_diff
+
+
+def score_angle_sector_differences(angle1: float, angle2: float) -> float:
+    """Helper for 'score_bearing which scores the angle difference
+
+    Args:
+        angle1, angle2:
+            The values are expected in degrees.
+    Returns:
+        The similarity of the sectors of the angles, from 1.0 (same sector) to 0.0 (16 sectors difference)
+    """
+
+    sector_diff = angle_sector_difference(angle1, angle2)
+    return 1.0 - (sector_diff / 16)
```

Comment:
I think it makes sense to give a candidate a maximum score if it lies anywhere within the sector communicated in the message.  But I think a linear scoring function should be applied for angles lying outside the sector.  Say, for example, the OpenLR message indicated sector 1 (0°-11.25°).  If you had 2 candidates, one with angle 12°, the other with angle 22°, both would be scored the same, even though the first candidate is closer to sector 1.